drgn/helpers/memory.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+"""
+Memory
+------
+
+The ``drgn.helpers.memory`` module provides helpers for working with memory and addresses.
+"""
+
+import operator
+from typing import Optional, Union, overload
+
+import drgn
+from drgn import IntegerLike, Object, Program, SymbolKind
+from drgn.helpers import escape_ascii_string
+from drgn.helpers.linux.slab import find_containing_slab_cache
+
+__all__ = ("identify_address",)
+
+
+SYMBOL_KIND_STR = {
+    SymbolKind.UNKNOWN: "symbol",
+    SymbolKind.OBJECT: "symbol object",
+    SymbolKind.FUNC: "symbol function",
+    SymbolKind.SECTION: "symbol section",
+    SymbolKind.FILE: "symbol file",
+    SymbolKind.COMMON: "symbol common",
+    SymbolKind.TLS: "symbol tls",
+    SymbolKind.IFUNC: "symbol ifunc",
```

Comment:
Actually, I think only object and function symbols deserve special treatment; the rest of the kinds are really obscure. You can delete the rest, then use `_SYMBOL_KIND_STR.get(symbol.kind, "symbol")` instead of `_SYMBOL_KIND_STR[symbol.kind]` so that anything else gets the default `"symbol"`.

```suggestion
    SymbolKind.OBJECT: "object symbol",
    SymbolKind.FUNC: "function symbol",
```

Reply:
Fixed.